org.eclipse.jdt.core/dom/org/eclipse/jdt/core/dom/ImplicitTypeDeclaration.java

```diff
@@ -161,4 +173,43 @@ final int internalGetSetIntProperty(SimplePropertyDescriptor property, boolean g
 		return super.internalGetSetIntProperty(property, get, value);
 	}
 
+	// from AbstractTypeDeclaration
+
+	@Override
+	public SimpleName getName() {
+		if (this.typeName == null) {
+			preLazyInit();
```

Comment:
I should have looked at the javadoc of preLazyInit() (did you? :smile: ): 
the pattern should always be enclosing in `synchronized(this)`.